client/app/account/billing/history/billing-history.controller.js

```diff
@@ -208,27 +220,44 @@ angular.module("Billing.controllers").controller("Billing.controllers.History",
             });
     }
 
-    (function init (_self) {
+    this.$onInit = function () {
+        this.isLoading = true;
         BillingUser.isVATNeeded().then((result) => {
-            _self.isVATNeeded = result;
+            this.isVATNeeded = result;
         });
+        this.canSetInvoiceByMail = false;
 
         loadDefaultPaymentMethod();
 
-        BillingDebtAccount.getDebtAccount()
-            .then((debtAccount) => {
-                _self.debtAccount = debtAccount;
-                if (_self.debtAccount.active) {
-                    _self.colSpan = COL_SPAN_DEBT_ACCOUNT;
-                }
-            })
-            .catch((err) => {
-                if (err.status === 404) {
-                    return null;
+        return $q.all([
+            BillingDebtAccount.getDebtAccount()
+                .then((debtAccount) => {
+                    this.debtAccount = debtAccount;
+                    if (this.debtAccount.active) {
+                        this.colSpan = COL_SPAN_DEBT_ACCOUNT;
+                    }
+                })
+                .catch((err) => err),
```

Comment:
you only need to catch once after $q.all otherwise the error will never be catched

catch((err) => err) will never raise an exception and always be resolved